### PR TITLE
Refactor Configuration (Simplify Config Part 3)

### PR DESCRIPTION
**Pull request by nupplaphil**
This is a FollowUp of #7371 

- Move Adapter business logic to Configuration
- Remove all Adapter
- Bugfixing Preload Configuration again (instead of #6688 )

I'll remove the draft once #7371 is merged

- Tested on local node
- Tested with installer 
- Tested with JIT (didn't retrieve all system config values of the DB) 
- Tested with Preload (retrieved all system config values of the DB)
- added tests to ensure that all config values will get loaded at once with preload configuration

The fourth - and last - PR will fix the PConfiguration like the Configuration

### CODE REVIEW

File: src/Core/Config/Cache/ConfigCache.php [hunk 1]
```diff
@@ -68,7 +68,7 @@ public function load(array $config, $overwrite = false)
 	 *
 	 * @return null|mixed Returns the value of the Config entry or null if not set
 	 */
-	public function get($cat, $key = null)
+	public function get(string $cat, $key = null)
```
[the remark this answers is not in the document]
Comment (nupplaphil) [reply src/Core/Config/Cache/ConfigCache.php]: Is a string typed parameter nullable? I wasn't sure and didn't want to get a null-fatal-error ...

[the remark this answers is not in the document]
Comment (nupplaphil) [reply src/Core/Config/Cache/ConfigCache.php]: It works ;-) ... Added a explicit test case to prove it ;-)

**Comment by tobiasd** [comment]
I had to revert the local git repo on the server to the point before the merge of this PR to get a working node back. No error messages as far as I could found.

Did you drop the support of legacy config files? Because I had no time to convert from the old file format so far (and in the near future).

**Comment by nupplaphil** [comment]
No, old formats should work too. What are you missing?

**Comment by nupplaphil** [comment]
Ah, I think I know the problem. I load the config-files and set them as 'loaded', so db values aren' t used in case there is a correspind config value.

Do you have a 'old' config file entry which is overwritten by a db value? I'll add a fixing PR in the next hour

**Comment by tobiasd** [comment]
Error page "The system is not available at the moment" which points to no DB connection. No error messages I the log that I could find. As the DB was running fine I reverted this PR and the system was working again.

**Comment by nupplaphil** [comment]
Plz can you try setting `system.config_adapter = preload` in your local config setting syntax and post if this works?

**Comment by nupplaphil** [comment]
@tobiasd should be fixed once #7384 is merged

**Comment by tobiasd** [comment]
> Plz can you try setting `system.config_adapter = preload` in your
> local config setting syntax and post if this works?

I hope that I can have a look next weekend, maybe even on Friday.

**Comment by tobiasd** [comment]
> Do you have a 'old' config file entry which is overwritten by a db
> value? I'll add a fixing PR in the next hour

Most likely. Initial theme selection comes to my mind and basic config stuff from the installer. I have not touched the config file for years. Well maybe not years, but even when I did the one conversion I did not change the entries from the htconfig values so they are not changed for years in the config file, and why bother, the DB values win.

**Pull request merged**